src/edu/psu/sweng500/team8/gui/SudokuGUI.java

```diff
@@ -51,6 +52,15 @@
 	private GameSession gameSession;
 	private static final String WIN_MESSAGE = "You won! Start a new game to play again.";
 
+	private boolean gameChanged = false;
```

Comment:
Suggest moving this to GameSession itself rather than the GUI. Then the event handler is not necessary and your unit test logic can be cleaned up a lot